File: C#.Net/JDI/Web/Selenium/Elements/Base/ClickableText.cs
using System;
using Epam.JDI.Core;
using Epam.JDI.Core.Interfaces.Common;
using Epam.JDI.Web.Selenium.Base;
using OpenQA.Selenium;

namespace Epam.JDI.Web.Selenium.Elements.Base
{
    public class ClickableText : Clickable, IText
    {
        public ClickableText() : this (null) { }
        public ClickableText(By byLocator = null, IWebElement webElement = null)
            : base(byLocator, webElement)
        {
        }

        protected Func<WebBaseElement, string> GetTextFunc =
            el => el.WebAvatar.FindImmediately(() => el.WebElement.Text, "");

        public string Value => Actions.GetValue(GetTextFunc);


        public string Text => Actions.GetText(GetTextFunc);

        public string WaitText(string text)
        {
            return Actions.WaitText(text, GetTextFunc);
        }

        public  string WaitMatchText(string regEx)
        {
            return Actions.WaitMatchText(regEx, GetTextFunc);
        }
    }
}

File: Python/JDI/web/selenium/elements/base/clickable_text.py
from JDI.web.selenium.elements.base.clickable import Clickable
from JDI.web.selenium.elements.common.text import Text


class ClickableText(Clickable, Text):

    def __init__(self, by_locator=None, web_element=None):
        if by_locator is not None:
            super(ClickableText, self).__init__(by_locator=by_locator)
        elif web_element is not None:
            super(ClickableText, self).__init__(web_element=web_element)
        else:
            super(ClickableText, self).__init__()
